src/pages/NewsPage/index.tsx

```diff
@@ -1,117 +1,168 @@
+	if (!response.ok) {
+		throw new Error('Failed to fetch news');
+	}
+	return response.json();
+};
+  
+const NewsComponentDrug: React.FC = () => {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+	const { data, isLoading, isError } = useQuery('drug', () => fetchNews('drug', '2024-03-29')) || {};
+
+	if (isLoading) {
+		return <div>Loading...</div>;
+	}
+
+	if (isError) {
+		return <div>Error: Failed to fetch news</div>;
+	}
+
+	return (
+		<div>
+		{/* eslint-disable-next-line */}
+		{data && data?.metadata.map((article: NewsItemProps) => (
+			<NewsItem
+			key={article.url}
+			title={article.title}
+			urlToImage={article.urlToImage}
+			url={article.url}
+			publishedAt={article.publishedAt}
+			/>
+		))}
+		</div>
+	);
+};
+  
+const NewsComponentVape: React.FC = () => {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+	const { data, isLoading, isError } = useQuery('vape', () => fetchNews('vape', '2024-03-29')) || {};
+
+	if (isLoading) {
+		return <div>Loading...</div>;
+	}
+
+	if (isError) {
+		return <div>Error: Failed to fetch news</div>;
+	}
+
+	return (	 
+		<div>
+		{/* eslint-disable-next-line */}
+		{data && data?.metadata.map((article: NewsItemProps) => (
+			<NewsItem
+			key={article.url}
+			title={article.title}
+			urlToImage={article.urlToImage}
+			url={article.url}
+			publishedAt={article.publishedAt}
+			/>
+		))}
+		</div>
+	);
+};
+  
+
 export default function NewsPage() {
-	const navigate = useNavigate();
-	const handleNavigateToDashboardPage = () => {
-		navigate("/dashboard");
+	const [selectedCategory, setSelectedCategory] = useState("drug");
+
+	const handleTabChange = (category: string) => {
+		setSelectedCategory(category);
 	};
+
 	return (
+		<QueryClientProvider client={queryClient}>
 		<div className="min-h-screen">
-			<div className="mb-12 mt-24 flex flex-col gap-4">
-				<div className="flex justify-center relative">
-					<IoIosArrowBack
-						className="text-3xl absolute left-0"
-						onClick={handleNavigateToDashboardPage}
-					/>
-					<header className="text-3xl font-bold text-center">
-						News
-					</header>
-				</div>
-				<div className="flex flex-row gap-10 items-center justify-center">
-					<Button title="Drugs" imageClassName="h-[105%] w-[105%]" />
-					<Button
-						title="Cyber Bully"
-						imageClassName="h-[105%] w-[105%]"
+			<div className="pb-12 pt-8 xl:pb-12 xl:pt-24 flex flex-col gap-8">
+				<div className="text-3xl xl:text-5xl font-bold text-center xl:text-center">News</div>
+				<div className="flex flex-row gap-2 xl:gap-10 justify-center">
+					<Button 
+						title="Drugs" 
+						onClick={() => handleTabChange("drug")} 
 					/>
-					<Button title="Vape" imageClassName="h-[105%] w-[105%]" />
-					<Button
-						title="Heritage"
-						imageClassName="h-[105%] w-[105%]"
+					<Button 

```

Comment:
would be nice if the button is disabled based off the category we have selected. Example, if we press drugs. The drugs button will be highlight and disabled, indicating that we are in that category